src/main/java/marquez/api/model/LineageEvent.java

```diff
@@ -0,0 +1,219 @@
+  public static class JobLink {
+    @NotNull public String namespace;
+    @NotNull public String name;
+  }
+
+  @Valid
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class LineageJob {
+
+    @NotNull public String namespace;
+    @NotNull public String name;
+    public JobFacet facets;
+  }
+
+  @Valid
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  @JsonPropertyOrder({"documentation", "sourceCodeLocation", "sql", "description"})
+  public static class JobFacet extends BaseFacet {
+
+    public DocumentationFacet documentation;
+    public SourceCodeLocationFacet sourceCodeLocation;
+    public SqlFacet sql;
+    private Map<String, Object> additional = new LinkedHashMap<>();
+
+    @JsonAnySetter
+    void setFacet(String key, Object value) {
+      additional.put(key, value);
+    }
+
+    @JsonAnyGetter
+    public Map<String, Object> getAdditionalFacets() {
+      return additional;
+    }
+  }
+
+  @Valid
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class DocumentationFacet extends BaseFacet {
+    @NotNull public String description;
+  }
+
+  @Valid
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class SourceCodeLocationFacet extends BaseFacet {
+
+    public String type;
+    public String url;
+  }
+
+  @Valid
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class SqlFacet extends BaseFacet {
+    @NotNull public String query;
+  }
+
+  @Valid
+  @ToString
+  @NotNull
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class LineageDataset {
+
+    @NotNull public String namespace;
+    @NotNull public String name;
+    public DatasetFacet facets;
+  }
+
+  @Valid
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  @JsonPropertyOrder({"documentation", "schema", "dataSource", "description"})
+  public static class DatasetFacet {
+
+    public DocumentationFacet documentation;
+    public SchemaFacet schema;
+    public DataSourceFacet dataSource;
+    public String description;
+    private Map<String, Object> additional = new LinkedHashMap<>();
+
+    @JsonAnySetter
+    void setFacet(String key, Object value) {
+      additional.put(key, value);
+    }
+
+    @JsonAnyGetter
+    public Map<String, Object> getAdditionalFacets() {
+      return additional;
+    }
+  }
+
+  @Valid
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class SchemaFacet extends BaseFacet {
+
+    public List<SchemaField> fields;
+  }
+
+  @Valid
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class SchemaField {
+
+    @NotNull public String name;
+    @NotNull public String type;
+    public String description;
+  }
+
+  @Valid
+  @ToString
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  public static class DataSourceFacet extends BaseFacet {
+
+    public String name;
+    public String uri;
+  }
+}
```

Comment:
_Minor_: I think the `Lineage` prefix may be redundant for the inner classes in the event. For example, `LineageEvent.LineageDataset` and `LineageEvent.Dataset` both convey that the a dataset is part of the event.